client/components/themes-list/index.jsx

```diff
@@ -325,7 +325,7 @@ function Options( { isFSEActive, recordTracksEvent, searchTerm, translate, upsel
 					destination: 'upload-theme',
 				} ),
 			url: isAtomic
-				? `https://${ selectedSite.slug }/wp-admin/theme-install.php`
+				? `${ selectedSite.options.admin_url }theme-install.php`
```

Comment:
Can we use the `getSiteAdminUrl()` selector instead? 

Furthermore, there is the possibility that the `admin_url` option doesn't exist, so let's make sure we've covered that scenario too. Using the selector can help a bit - it will already return `null` in that scenario.